core/liquidity/v2/sla.go

```diff
@@ -80,26 +96,41 @@ func (e *Engine) CalculateSLAPenalties(now time.Time) {
 		// if LP meets commitment
 		// else LP does not meet commitment
 		if timeBookFraction.LessThan(e.commitmentMinTimeFraction) {
-			feePenalty = num.DecimalOne()
+			feePenalty = one
 			bondPenalty = e.calculateBondPenalty(timeBookFraction)
 		} else {
 			feePenalty = e.calculateCurrentFeePenalty(timeBookFraction)
 			bondPenalty = num.DecimalZero()
 		}
 
 		commitment.allEpochsPenalties = append(commitment.allEpochsPenalties, feePenalty)
+
+		// drop unnecessary historical penalties
+		if len(commitment.allEpochsPenalties) > maximumNumberOfHistoricalPenalties {
+			commitment.allEpochsPenalties = commitment.allEpochsPenalties[:maximumNumberOfHistoricalPenalties]
+		}
+
 		previousPenalties := commitment.allEpochsPenalties[:len(commitment.allEpochsPenalties)-1]
 
 		penaltiesPerParty[party] = &SlaPenalty{
 			Bond: bondPenalty,
 			Fee:  e.calculateHysteresisFeePenalty(feePenalty, previousPenalties),
 		}
+
+		if penaltiesPerParty[party].Fee.Equal(one) {
+			partiesWithFullFeePenaltyCount++
+		}
+	}
+
+	e.slaPenalties = SlaPenalties{
```

Comment:
do we actually need to store it for future use? I don't see where it's used beyond as the return value. There's an API below that returns it but nothing is currently using it so not sure.